audio support

soumith opened the pull request:
display.audio is supported now.

Test:
th -e "require 'audio';disp=require 'display'; disp.audio(audio.samplevoice(), {rate=22000});"

<img width="387" alt="screen shot 2015-12-30 at 3 58 42 am" src="https://cloud.githubusercontent.com/assets/1310570/12048485/a6fdf490-aea9-11e5-9ca8-91e8df29c7fc.png">

szym commented on init.lua:
```diff
@@ -166,5 +166,60 @@ function M.text(txt, opts)
+      local sampleRate = opts.rate or 44100 -- default sample rate
+      -- use temporary file
+      fname = os.tmpname() .. '.ogg'
+      audio.save(fname, data, sampleRate)
```
Is there a way to avoid writing it to file? Similar to `image.compressJPG` above?

soumith replied:
I've added audio.compress / decompress to the audio package today. However, it is only supported on Linux OOB because it uses fmemopen.
I added a commit here to use that code path where possible, and fallback to temporary files.

soumith commented:
I've added audio.compress / decompress to the audio package today. However, it is only supported on Linux OOB because it uses fmemopen.
I added a commit here to use that code path where possible, and fallback to temporary files.
